src/classkit.cs

```diff
@@ -0,0 +1,329 @@
+		Parent,
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,3)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface CLSObject : NSSecureCoding {
+
+		[Export ("dateCreated")]
+		NSDate DateCreated { get; }
+
+		[Export ("dateLastModified")]
+		NSDate DateLastModified { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,3)]
+	[BaseType (typeof (CLSObject))]
+	[DisableDefaultCtor]
+	interface CLSActivity {
+
+		[Export ("progress")]
+		double Progress { get; set; }
+
+		[Export ("duration")]
+		double Duration { get; }
+
+		[Export ("primaryActivityItem", ArgumentSemantic.Assign)]
+		CLSActivityItem PrimaryActivityItem { get; set; }
+
+		[Export ("addProgressRangeFromStart:toEnd:")]
+		void AddProgressRange (double start, double end);
+
+		[Export ("addAdditionalActivityItem:")]
+		void AddAdditional (CLSActivityItem activityItem);
+
+		[Export ("additionalActivityItems")]
+		CLSActivityItem [] AdditionalActivityItems { get; }
+
+		// From CLSActivity (Activation) Category
+
+		[Export ("started")]
+		bool Started { [Bind ("isStarted")] get; }
+
+		[Export ("start")]
+		void Start ();
+
+		[Export ("stop")]
+		void Stop ();
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,3)]
+	[BaseType (typeof (CLSObject))]
+	[DisableDefaultCtor]
+	interface CLSActivityItem {
+
+		[Export ("title")]
+		string Title { get; set; }
+
+		[Export ("identifier")]
+		string Identifier { get; }
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,3)]
+	[BaseType (typeof (CLSActivityItem))]
+	[DisableDefaultCtor]
+	interface CLSBinaryItem {
+
+		[Export ("value")]
+		bool Value { get; set; }
+
+		[Export ("valueType")]
+		CLSBinaryValueType ValueType { get; }
+
+		[Export ("initWithIdentifier:title:type:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string identifier, string title, CLSBinaryValueType valueType);
+	}
+
+	[NoWatch, NoTV, NoMac, iOS (11,3)]
+	[BaseType (typeof (CLSObject))]
+	[DisableDefaultCtor]
+	interface CLSContext {
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[NoWatch, NoTV, NoMac, iOS (11,4)]
+		[NullAllowed, Export ("universalLinkURL", ArgumentSemantic.Assign)]
+		NSUrl UniversalLinkUrl { get; set; }
+
+		[Export ("type")]
+		CLSContextType Type { get; }
+
+		[Export ("title")]
+		string Title { get; set; }
+
+		[Export ("displayOrder")]
+		nint DisplayOrder { get; set; }
+
+		[Protected]
+		[NullAllowed, Export ("topic")]
+		NSString WeakTopic { get; set; }
+
+		[Export ("initWithType:identifier:title:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CLSContextType type, string identifier, string title);
+
+		[Export ("active")]
+		bool Active { [Bind ("isActive")] get; set; }
+
+		[Export ("becomeActive")]
+		void BecomeActive ();
+
+		[Export ("resignActive")]
+		void ResignActive ();
+
+		// From CLSContext (Hierarchy) Category
+
+		[NullAllowed, Export ("parent")]
+		CLSContext Parent { get; }
+
+		[Export ("removeFromParent")]
+		void RemoveFromParent ();
+
+		[Export ("addChildContext:")]
+		void AddChild (CLSContext childContext);
```

Comment:
Just `Add`?

Reply:
Same reasoning, `CLSContext` does not refer to be a collection, I think `AddChild` es explicit on its intent

Reply:
Fair enough. Let's leave it as is them.